### ISSUE TITLE
Conflict resolution doesn't drop copy local assemblies which were absorbed into .NETStandard2.1

### ISSUE THREAD

**Issue by ericstj** [dotnet/sdk#18129]
Build the following project:
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netstandard2.1</TargetFramework>
    <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="System.Security.Cryptography.Pkcs" Version="5.0.1" />
  </ItemGroup>
</Project>
```

Expect: 
System.Buffers.dll, System.Memory.dll, System.Numerics.Vectors.dll should be absent from the output directory.

Actual:
All are present.

These are all conflict resolved from the reference assets, but not runtime.  It looks like netstandard is missing runtime handling for conflict resolution.  Looks like the same thing happened during `netstandard2.0` so this isn't new.

Perhaps we could avoid this by leveraging https://github.com/dotnet/sdk/blob/a30e465a2e2ea4e2550f319a2dc088daaafe5649/src/Tasks/Common/ConflictResolution/ResolvePackageFileConflicts.cs#L35-L44

Might improve perf on netstandard as well.

**Comment by ericstj** [comment]
I was able to workaround this with the following:
```

  <Target Name="IncludeNetStandardConflictResolution" BeforeTargets="_HandlePackageFileConflicts">
    <ItemGroup>
      <_RuntimeAssetsForConflictResolution Include="@(Reference)" Condition="'%(Reference.NuGetPackageId)' == 'NETStandard.Library' or '%(Reference.NuGetPackageId)' == 'NETStandard.Library.Ref'" />
    </ItemGroup>
  </Target>
```

My suggestion of using `PackageOverrides` isn't good enough, since it looks like that's not considered *unless* there is a conflict. So the fix here would need to represent the runtime items expected by a `.NETStandard`-implementing framework.  To avoid hitting files, it means we should probably have a platform manifest or framework list.

**Comment by dsplaisted** [comment]
Is `CopyLocalLockFileAssemblies` really even valid for a .NET Standard project?  Doesn't it basically mean the implementation assemblies should be copied to the output folder?  But on .NET Standard there's no guarantee that there are implementation assemblies at all (for .NET Standard).

**Comment by ericstj** [comment]
Right, but people still do it, and in some places we tell customers to do it (EG: roslyn analyzers).   We could block it completely, but I suspect that's too breaking.  If we allow it, then we should fix this.
